### PR TITLE
[GPU] Added Pad/Interpolate dynamic support and MVN dynamic tests

### PR DESCRIPTION
**Pull request by Lyamin-Roman**

### CODE REVIEW

File: src/plugins/intel_gpu/include/intel_gpu/primitives/resample.hpp [hunk 1]
```diff
@@ -123,6 +123,36 @@ struct resample : public primitive_base<resample> {
             throw std::runtime_error("Resample's scales/axes count does not match");
     }
 
+    /// @brief resample with non-constant scales
+    resample(const primitive_id& id,
+             const primitive_id& input,
+             const std::vector<int64_t>& sizes,
```
Comment (yeonbok) [inline src/plugins/intel_gpu/include/intel_gpu/primitives/resample.hpp:129]: isnt it better to have one unified primitive API with vector<primitive_id> inputs and then process each cases w.r.t. the # of inputs?

File: src/plugins/intel_gpu/include/intel_gpu/primitives/resample.hpp [hunk 2]
```diff
@@ -123,6 +123,36 @@ struct resample : public primitive_base<resample> {
...
+             const primitive_id& scales,
+             const std::vector<int64_t>& axes,
+             const std::vector<size_t>& pads_begin = {},
+             const std::vector<size_t>& pads_end = {},
+             int32_t antialias = 0,
+             float cube_coeff = -0.75f,
+             InterpolateOp::InterpolateMode operation_type = InterpolateOp::InterpolateMode::LINEAR,
+             InterpolateOp::ShapeCalcMode shape_calc_mode = InterpolateOp::ShapeCalcMode::SIZES,
```
Comment (yeonbok) [inline src/plugins/intel_gpu/include/intel_gpu/primitives/resample.hpp:137]: should'nt this be SCALES mode?

File: src/plugins/intel_gpu/src/graph/impls/ocl/resample.cpp [hunk 3]
```diff
@@ -150,8 +150,16 @@ struct resample_impl : typed_primitive_impl_ocl<resample> {
         us_params.pads_begin = convert_pads(primitive->pads_begin, dimsNum);
         us_params.pads_end = convert_pads(primitive->pads_end, dimsNum);
 
-        for (size_t i = 0; i < primitive->scales.size(); i++) {
-            us_params.axesAndScales[convert_axis(primitive->axes[i], dimsNum)] = primitive->scales[i];
+        auto scales = primitive->scales;
+        bool scales_calc_mod = primitive->shape_calc_mode == resample::InterpolateOp::ShapeCalcMode::SCALES;
```
Comment (ahnyoung-paul) [inline src/plugins/intel_gpu/src/graph/impls/ocl/resample.cpp:154]: This condition checks whether `shape_calc_mode` is scales mode. but this commit only sets for size mode as follows: 

`InterpolateOp::ShapeCalcMode shape_calc_mode = InterpolateOp::ShapeCalcMode::SIZES`

When the `shape_calc_mode` will be set to scale mode?

Comment (Lyamin-Roman) [reply src/plugins/intel_gpu/src/graph/impls/ocl/resample.cpp:154]: shape_calculation_mode is set when the primitive constructor is called in CreateInterpolateOp() and contains the value from ngraph::op::v4::Interpolate op

File: src/plugins/intel_gpu/src/graph/impls/ocl/resample.cpp [hunk 4]
```diff
@@ -150,8 +150,16 @@ struct resample_impl : typed_primitive_impl_ocl<resample> {
...
+        if (scales_calc_mod && impl_param.input_layouts.size() == 2) {
+            auto mem = impl_param.memory_deps.at(1);
+            float* buffer = static_cast<float*>(mem->buffer_ptr());
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/impls/ocl/resample.cpp:157]: what if the second input is sizes_id so that it should be int32_t?

Comment (Lyamin-Roman) [reply src/plugins/intel_gpu/src/graph/impls/ocl/resample.cpp:157]: Fixed, in this case the code should not be called

**Comment by yeonbok** [comment]
Verified with real model.

**Review by yeonbok — APPROVED** [review]

**Pull request merged**